src/operator/nn/mkldnn/mkldnn_pooling-inl.h

```diff
@@ -0,0 +1,393 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file mkldnn_pooling-inl.h
+ * \brief
+*/
+#ifndef MXNET_OPERATOR_NN_MKLDNN_MKLDNN_POOLING_INL_H_
+#define MXNET_OPERATOR_NN_MKLDNN_MKLDNN_POOLING_INL_H_
+
+#if MXNET_USE_MKLDNN == 1
+
+#include <utility>
+#include <mkldnn.hpp>
+#include "../pooling-inl.h"
+#include "./mkldnn_base-inl.h"
+
+namespace mxnet {
+namespace op {
+
+class MKLDNNPoolingFwd {
+ public:
+  MKLDNNPoolingFwd(const mxnet::NDArray &input,
+                   const mxnet::NDArray &output,
+                   int kernel_h, int kernel_w,
+                   int stride_h, int stride_w,
+                   int padding_t, int padding_b, int padding_l, int padding_r,
+                   mkldnn::algorithm alg_kind,
+                   bool with_workspace, bool is_train) :
+                   _is_train(is_train),
+                   _with_workspace(with_workspace),
+                   _alg_kind(alg_kind),
+                   fwd(nullptr), data(nullptr), out(nullptr), workspace(nullptr) {
+    _Init(input, output,
```

Comment:
Why underscore?

Reply:
Intended to indicate this is a private method. I have searched in mxnet and google coding style, seems there is no special prefix or suffix for private method, but a underscore suffix for private variable. I will change code to align with that rule.